### PR TITLE
fix: Internal Transfer Material Request cycle and tracking fixed till purchase receipt

### PR DESCRIPTION
**Pull request by vishdha**
Problem statement:
1. User create Material Request -> Purchase Order. Material Request status get updated (Ordered)
2. Purchase Order (Internal)-> Sales Order(Internal) -> Delivery Note(Internal) -> Purchase Receipt
3. **Purchase Receipt get submitted still Material Request status remain same (Ordered) Instead it should be Received**
4. This fixed added link of material request and status get updated of MR(Received) once PR submitted 

https://user-images.githubusercontent.com/6947417/200765870-63e7340d-942c-4bcd-876b-195d7f53ec29.mp4

### CODE REVIEW

**Comment by vishdha** [comment]
@deepeshgarg007 @rohitwaghchaure  Please review and backport to V14 and v13

**Comment by vishdha** [comment]
@sagarvora @rohitwaghchaure @deepeshgarg007 Can you review this.

**Review by s-aga-r — CHANGES_REQUESTED** [review]
@vishdha Could you add a test case for the same?

**Comment by vishdha** [comment]
@s-aga-r Test case added for same

**Comment by s-aga-r** [comment]
> @s-aga-r Test case added for same

@vishdha Test is failing, please fix it.

**Comment by vishdha** [comment]
> 

@s-aga-r Test case fixed. Please review and merge this. its import to merge in version-14

File: erpnext/selling/doctype/sales_order_item/sales_order_item.json [hunk 1]
```diff
@@ -847,7 +849,18 @@
    "label": "quotation_item",
    "no_copy": 1,
    "read_only": 1
-  }
+  },
```
Comment (s-aga-r) [inline erpnext/selling/doctype/sales_order_item/sales_order_item.json:852]: @vishdha Don't edit the DocType JSON file manually.

**Review by s-aga-r — APPROVED** [review]

**Pull request merged**